start_control_ssm.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+log() {
+  echo "$*" >&2
+}
+
+enable_hybrid_env_ssm() {
+  # SSM parameters necessary to register with a hybrid activation
+  activation_code=$(get_user_data_ssm "activation-code")
+  activation_id=$(get_user_data_ssm "activation-id")
+  region=$(get_user_data_ssm "region")
+
+  # Register with AWS Systems Manager (SSM)
+  amazon-ssm-agent -register -code "${activation_code}" -id "${activation_id}" -region "${region}"
+  if [[ "$?" -eq 1 ]]; then
+    log "Failed to register with AWS Systems Manager (SSM)"
+    exit 1
+  fi
+}
+
+# Extract the values from user-data json, and exit on failure (if any)
+get_user_data_ssm() {
+  local param_value
+  local ssm_param="${1:?}"
+  if ! param_value=$(jq --arg ssm_param "${ssm_param}" -e -r '.["ssm"][$ssm_param]' "${USER_DATA}"); then
+    log "Failed to parse ${ssm_param} from ${USER_DATA}"
+    exit 1
+  fi
+  echo "${param_value}"
+}
+
+# If /.bottlerocket/host-containers/current/user-data exists and is not empty
+# and the symlinked /var/lib/amazon/ssm/registration file is not populated,
+# then check to see if the user-data file contains ssm at the top-level. If so,
+# attempt to manually register with SSM with a hybrid activation.
+declare -r PERSISTENT_STORAGE_BASE_DIR="/.bottlerocket/host-containers/current"
+declare -r USER_DATA="${PERSISTENT_STORAGE_BASE_DIR}/user-data"
+declare -r SSM_AGENT_PERSISTENT_STATE_DIR="${PERSISTENT_STORAGE_BASE_DIR}/ssm"
+declare -r SSM_AGENT_LOCAL_STATE_DIR="/var/lib/amazon/ssm"
+
+# Create the default directory structure on the persistent storage location
+mkdir -p "${SSM_AGENT_PERSISTENT_STATE_DIR}"
+chmod 750 "${SSM_AGENT_PERSISTENT_STATE_DIR}"
+
+if [[ -s "${USER_DATA}" ]] && [[ ! -s "${SSM_AGENT_LOCAL_STATE_DIR}/registration" ]]; then
+  if jq -e -r '.["ssm"]' "${USER_DATA}" &>/dev/null; then
+    enable_hybrid_env_ssm
+  fi
```

Comment:
```suggestion
if [[ -s "${USER_DATA}" ]] \
&& [[ ! -s "${SSM_AGENT_LOCAL_STATE_DIR}/registration" ]] \
&& jq --exit-status '.ssm' "${USER_DATA}" &>/dev/null \
; then
  enable_hybrid_env_ssm
```
* nit on layout to make the list of conditions clearer
* nit on spelling out `-e` so we don't have to look it up as often
* nit on simplifying jq condition
* don't think we need `-r` since we're throwing out the output anyway